packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackset-types.ts

```diff
@@ -0,0 +1,413 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * The source of a StackSet template
+ */
+export abstract class StackSetTemplate {
+  /**
+   * Use a file in an artifact as Stack Template.
+   */
+  public static fromArtifactPath(artifactPath: codepipeline.ArtifactPath): StackSetTemplate {
+    return new class extends StackSetTemplate {
+      public readonly artifactsReferenced?: codepipeline.Artifact[] | undefined = [artifactPath.artifact];
+
+      public render() {
+        return artifactPath.location;
+      }
+    }();
+  }
+
+  /**
+   * Which artifacts are referenced by this template
+   *
+   * Does not need to be called by app builders.
+   */
+  public abstract readonly artifactsReferenced?: codepipeline.Artifact[] | undefined;
+
+  /**
+   * Render the template to the pipeline
+   *
+   * Does not need to be called by app builders.
+   */
+  public abstract render(): any;
+}
+
+/**
+ * Where Stack Instances will be created from the StackSet
+ */
+export abstract class StackInstances {
+  /**
+   * Create stack instances in a literal set of accounts or organizational units, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromList(targets: string[], regions: string[]): StackInstances {
+    if (targets.length === 0) {
+      throw new Error("'targets' may not be an empty list");
+    }
+
+    if (regions.length === 0) {
+      throw new Error("'regions' may not be an empty list");
+    }
+
+    return new class extends StackInstances {
+      public bind(_scope: Construct): StackInstancesBindResult {
+        return {
+          stackSetConfiguration: {
+            DeploymentTargets: targets.join(','),
+            Regions: regions.join(','),
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * Create stack instances in a set of accounts or organizational units taken from the pipeline artifacts, and a set of regions
```

Comment:
Can we explain what format in the file do the targets have to take here? Kind of like we did with `fromArtifactPath()` in `StackSetParameters` below? Those were some awesome docs, would be great to have them here too.